gui/wxpython/mapdisp/properties.py

```diff
@@ -229,14 +230,59 @@ def _onToggleCheckBox(self, event):
             self.giface.updateMap.emit(render=False)
 
 
+class ChBProjection:
+    """Checkbox to enable user defined projection"""
+
+    def __init__(self, parent, statusBarProperties):
+        self.properties = statusBarProperties
+        self.name = "projection"
+        self.label = _("Projection")
+        self.defaultLabel = _("Use defined projection")
+        self.widget = wx.CheckBox(parent=parent, id=wx.ID_ANY, label=self.defaultLabel)
+        self.widget.SetToolTip(
+            wx.ToolTip(
+                _(
+                    "Reproject coordinates displayed "
+                    "in the statusbar. Projection can be "
+                    "defined in GUI preferences dialog "
+                    "(tab 'Projection')"
+                )
+            )
+        )
+        self.widget.Bind(wx.EVT_CHECKBOX, self._onToggleCheckBox)
+
+    def GetValue(self, value):
+        self.widget.GetValue(value)
+
+    def GetWidget(self):
+        """Returns underlying widget.
+
+        :return: widget or None if doesn't exist
+        """
+        return self.widget
+
+    def _onToggleCheckBox(self, event):
+        self.properties.useDefinedProjection = self.widget.GetValue()
+        epsg = self.properties.epsg
+        if epsg:
+            label = "%s (EPSG: %s)" % (self.defaultLabel, epsg)
+            self.widget.SetLabel(label)
+        else:
+            self.widget.SetLabel(self.defaultLabel)
+
+        # disable long help

```

Comment:
Not sure how to deal with this method. Seems like quite dependent on mapframe. I have noticed that it is quite common in statusbar.py (SbItem, SbMapScale, SbGoTo and SbProjection), however we need to get rid of map display dependency. So maybe put it into giface?

Reply:
What is it even doing? Doesn't look very important.

Reply:
Seems to me the same. As far I was able to find out it sets the help to the statusbar item. E.g. if enabled it sets a tool long help something like: WindowIDRef: -31309 Zoom to default or saved region, save to named region, .... If not enabled the help is just WindowIDRef: -31309.